### ISSUE TITLE
Support AWS SSO profiles

### ISSUE THREAD

**Issue by iainelder** [peak/s5cmd#385]
I set up my environment variables and check that the session is valid using the AWS CLI.

```
$ export AWS_PROFILE=sandbox-logging
$ export AWS_DEFAULT_REGION=eu-west-1
$ aws sts get-caller-identity
{
    "UserId": "AROAXXXXXXXXXXXXXXXXX:iain",
    "Account": "111111111111",
    "Arn": "arn:aws:sts::111111111111:assumed-role/AWSReservedSSO_AdministratorAccess_aaaaaaaaaaaaaaaa/iain"
}
```

s5cmd seems to ignore my environment variables and instead tries to query the EC2 metadata service.

```
$ s5cmd --log debug ls
DEBUG retryable error: RequestError: send request failed
caused by: Put "http://169.254.169.254/latest/api/token": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
DEBUG retryable error: RequestError: send request failed
caused by: Put "http://169.254.169.254/latest/api/token": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
DEBUG retryable error: RequestError: send request failed
caused by: Put "http://169.254.169.254/latest/api/token": dial tcp 169.254.169.254:80: connect: no route to host
```

I was expecting output like this:

```
$ aws s3 ls
2021-11-23 15:29:46 aws-sam-cli-managed-default-samclisourcebucket-xxxxxxxxxxxx
2021-11-16 17:47:47 cf-templates-xxxxxxxxxxxxx-eu-west-1
2021-11-30 20:11:58 org-trail-xxxxxxxx
2021-12-01 12:36:42 org-trail-yyyyyyyy
...
```

I'm using an SSO profile. Does that matter?

```cfg
[profile sandbox-logging]
sso_start_url = https://d-1111111111.awsapps.com/start
sso_region = eu-west-1
sso_account_id = 111111111111
sso_role_name = AdministratorAccess
region = eu-west-1
```

**Comment by iainelder** [comment]
> I'm using an SSO profile. Does that matter?

When I generate temporary credentials for the SSO profile using [aws2-wrap](https://github.com/linaro-its/aws2-wrap), s5cmd works just fine.

```
$ s5cmd --log debug ls
2021/11/23 14:29:46  s3://aws-sam-cli-managed-default-samclisourcebucket-xxxxxxxxxxxx
2021/11/16 16:47:47  s3://cf-templates-xxxxxxxxxxxxx-eu-west-1
2021/11/30 19:11:58  s3://org-trail-xxxxxxxx
2021/12/01 14:25:45  s3://org-trail-yyyyyyyy
...
```

To support SSO profiles, s5cmd needs to upgrade its dependency on aws-sdk-go from v1.35.13.

https://github.com/peak/s5cmd/blob/fd03f6e870d09e9ecc2565f7751fefe366297134/go.mod#L6

The minimum version that supports SSO profiles is v.1.37.0.

The latest version to improve support for SSO profiles was v1.38.42.

Would you be able to upgrade the dependency to make this work automatically?

**Comment by igungor** [comment]
> The minimum version that supports SSO profiles is [v.1.37.0](https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md#release-v1370-2021-01-28).
>
> The latest version to improve support for SSO profiles was [v1.38.42](https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md#release-v13842-2021-05-18).
>
> Would you be able to upgrade the dependency to make this work automatically?

Thanks for the pointer. We've updated the SDK version to `v1.40.25`. Could you please try it out and see if it works for you?

**Comment by iainelder** [comment]
Thanks @igungor . I won't have time to try it until next week, but I'll let you know how it works for me when I have a chance.

**Comment by igungor** [comment]
Hi @iainelder 

We've released [v2.0.0-beta](https://github.com/peak/s5cmd/releases/tag/v2.0.0-beta) which includes the change you requested along with new features and bugfixes.

**Comment by iainelder** [comment]
@igungor , thanks! Verstion v2.0.0-beta-df6b29f is working as expected.

**Comment by anthonymobile** [comment]
similar issues here, also resolved with aws2-wrap but not working in v2.0.0-beta

```
> aws --version
aws-cli/2.7.7 Python/3.9.11 Darwin/21.5.0 exe/x86_64 prompt/off
> s5cmd version
v2.0.0-beta.2-c978e4a
> aws sso login
Attempting to automatically open the SSO authorization page in your default browser.
If the browser does not open or you wish to use a different device to authorize this request, open the following URL:

https://device.sso.us-east-1.amazonaws.com/

Then enter the code:

XCPM-BGWB
Successfully logged into Start URL: https://cornell-sso.awsapps.com/start
> s5cmd ls
ERROR "ls": InvalidAccessKeyId: The AWS Access Key Id you provided does not exist in our records. status code: 403, request id: Y9HYYSV1C0D2ZF4Z, host id: fH3IQlAlCpTYABAwTKwFQupUIp+EpMg2aGieepRpN9pQfp8FBcnWATPWQge70U2EFpgXk12mvW0=
> aws s3 ls
2022-05-04 16:25:18 aws-cloudtrail-logs-management-events-870747888580-e6148274
2022-05-06 01:15:16 aws-sam-cli-managed-default-samclisourcebucket-1ll3d7xend2xd
> aws2-wrap s5cmd ls
2022/05/04 20:25:18  s3://aws-cloudtrail-logs-management-events-870747888580-e6148274
2022/05/06 05:15:16  s3://aws-sam-cli-managed-default-samclisourcebucket-1ll3d7xend2xd
/Users/anthonytownsend/.aws:
```

**Comment by ollie-bell** [comment]
Please can the readme be updated with instructions on using s5cmd with AWS SSO profiles? I'm unclear on how to get s5cmd to work whilst logged in with an AWS SSO profile (i.e. followed instructions here https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sso.html). Thanks
